ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -84,8 +102,46 @@ export default function SignatureRequestSIWE({
     [cancelPersonalMessage],
   );
 
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+
+  const conversionRateFromSelector = useSelector(conversionRateSelector);
+
+  const conversionRate = useNativeCurrencyAsPrimaryCurrency
+    ? null
+    : conversionRateFromSelector;
+
+  const balanceInBaseAsset = conversionRate
+    ? formatCurrency(
+        getValueFromWeiHex({
+          value: fromAccount.balance,
+          fromCurrency: nativeCurrency,
+          toCurrency: currentCurrency,
+          conversionRate,
+          numberOfDecimals: 6,
+          toDenomination: EtherDenomination.ETH,
+        }),
+        currentCurrency,
+      )
+    : new Numeric(fromAccount.balance, 16, EtherDenomination.WEI)
+        .toDenomination(EtherDenomination.ETH)
+        .round(6)
+        .toBase(10)
+        .toString();
+
+  const currentNetwork = getNetworkName(provider, t);
+
   return (
     <div className="signature-request-siwe">
+      <NetworkAccountBalanceHeader
+        networkName={currentNetwork}
+        accountName={fromAccount.name}
+        accountBalance={balanceInBaseAsset}
+        tokenName={
+          conversionRate ? currentCurrency?.toUpperCase() : nativeCurrency
+        }
+        accountAddress={fromAccount.address}
+        chainId={currentChainId}
+      />
```

Comment:
what do you think about moving the logic into the Header component from './signature-request-siwe-header' to maintain separations of concerns?